### ISSUE TITLE
Generating empty map when supplied a map with a single default value

### ISSUE THREAD

**Issue by kika** [tokio-rs/prost#698]
Please have a look at https://github.com/google/protobuf.dart/issues/719
The message that makes Dart frontend explode is generated by prost and Google team thinks it could be a bug on the prost side too (the fact that Dart throws exception is a bug of itself). 
If this is a valid assumption and their read of protobuf spec is assumed correct here, I'll work on the repro case (and maybe PR, I'm far from being an expert in protobuf and even less so in Rust).

**Comment by LucioFranco** [comment]
Yeah, I would be happy to review a repro/PR. Thanks for opening this!

**Comment by ldm0** [comment]
Related: https://github.com/tokio-rs/prost/issues/99
